src/MagnumPlugins/AssimpImporter/Test/Test.cpp

```diff
@@ -0,0 +1,257 @@
+    CORRADE_VERIFY(mesh);
+    CORRADE_VERIFY(mesh->isIndexed());
+    CORRADE_COMPARE(mesh->primitive(), MeshPrimitive::Triangles);
+    CORRADE_COMPARE(mesh->positionArrayCount(), 1);
+    CORRADE_COMPARE(mesh->normalArrayCount(), 1);
+    CORRADE_COMPARE(mesh->textureCoords2DArrayCount(), 0);
+    CORRADE_COMPARE(mesh->colorArrayCount(), 0);
+
+    CORRADE_COMPARE(mesh->positions(0), (std::vector<Vector3>{
+            {-1.0f, 1.0f, 1.0f}, {-1.0f, -1.0f, 1.0f}, {1.0f, -1.0f, 1.0f}}));
+    CORRADE_COMPARE(mesh->indices(), (std::vector<UnsignedInt>{0, 1, 2}));
+
+    std::unique_ptr<Trade::ObjectData3D> meshObject = importer.object3D(1);
+    CORRADE_COMPARE(meshObject->instanceType(), ObjectInstanceType3D::Mesh);
+    CORRADE_COMPARE(meshObject->instance(), 0);
+}
+
+void AssimpImporterTest::scene() {
+    AssimpImporter importer;
+    CORRADE_VERIFY(importer.openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR, "scene.dae")));
+
+    CORRADE_COMPARE(importer.sceneCount(), 1);
+    CORRADE_COMPARE(importer.defaultScene(), 0);
+    std::optional<Trade::SceneData> data = importer.scene(0);
+    CORRADE_VERIFY(data);
+
+    CORRADE_COMPARE(data->children2D(), {});
+    CORRADE_COMPARE(data->children3D(), {0});
+
+    std::unique_ptr<Trade::ObjectData3D> rootObject = importer.object3D(0);
+    CORRADE_COMPARE(rootObject->children(), {1});
+    //CORRADE_COMPARE(rootObject->instanceType(), ObjectInstanceType3D::Mesh); // Should be Empty, but Assimp creates mesh for root node for some reason
+    CORRADE_COMPARE(rootObject->transformation(),
+                    Matrix4({1.0f,  0.0f, 0.0f, 0.0f},
+                            {0.0f,  0.0f, 1.0f, 0.0f},
+                            {0.0f, -1.0f, 0.0f, 0.0f},
+                            {0.0f,  0.0f, 0.0f, 1.0f}));
+
+    std::unique_ptr<Trade::ObjectData3D> explicitRootObject = importer.object3D(1);
+    CORRADE_COMPARE(explicitRootObject->children(), {2});
+    CORRADE_COMPARE(explicitRootObject->instanceType(), ObjectInstanceType3D::Empty);
+    CORRADE_COMPARE(explicitRootObject->transformation(), Matrix4());
+
+    std::unique_ptr<Trade::ObjectData3D> childObject = importer.object3D(2);
+    CORRADE_COMPARE(childObject->transformation(),
+                    Matrix4({0.813798f, 0.469846f, -0.34202f, 0.0f},
+                            {-0.44097f, 0.882564f, 0.163176f, 0.0f},
+                            {0.378522f, 0.0180283f, 0.925417f, 0.0f},
+                            {1.0f, 2.0f, 3.0f, 1.0f}).transposed());
+}
+
+void AssimpImporterTest::texture() {
+    AssimpImporter importer;
+    CORRADE_VERIFY(importer.openFile(Utility::Directory::join(ASSIMPIMPORTER_TEST_DIR, "embedded_texture.blend")));
+
+    CORRADE_COMPARE(importer.textureCount(), 1);
+    CORRADE_COMPARE(importer.image2DCount(), 1);
+
+    CORRADE_SKIP("This test requires using AssimpImporter via a plugin manager which has access to PngImporter for blender embedded image data.");
```

Comment:
Shouldn't be a problem to manage -- look in the OGEX importer tests how it's done.